### ISSUE TITLE
A4 Page Size leaving white edge when printed to PDF

### ISSUE THREAD

**Issue by Gazook89** [naturalcrit/homebrewery#2716]
### Renderer

v3

### Browser

Chrome

### Operating System

MacOS

### What happened?

When using the A4 Page Size snippet, and printing to PDF (setting the print dialog page size to A4), the resulting PDF has a ~1px white edge along the bottom and right sides.  It also adds a partially transparent black border along the edge of the background image.  I'm pretty sure that edge isn't part of the background image itself.

Increasing the page size, such as `height`, in the brew snippet does not help.  In the following image, you can see that the brew's "page size" was increased and thus overflowed to the next page when sent to PDF---yet the white edge remains:

![image](https://user-images.githubusercontent.com/58999374/223296489-a0dd0cd3-3487-4e4b-a15d-225e12e2f975.png)

**This doesn't seem to be an issue with the US Letter page size.**  I have the issue on my Mac (using Chrome and Preview), and @G-Ambatte has confirmed as well, I'm pretty sure on Windows.  [Original report on reddit was also on Windows 11.](https://old.reddit.com/r/homebrewery/comments/11k10es/thin_white_bar_visible_after_exporting_a4_sized/)

Just some quick scanning of Chromium Monorail:

- [Bug 914817 - Print with "Save as PDF" adds the white line on the page bottom - Opened 2018](https://bugs.chromium.org/p/chromium/issues/detail?id=914817&q=pdf%20print%20white%20border&can=2)
- [Bug 1323671 - Page.printToPDF generates file with the white borders - Opened 2022](https://bugs.chromium.org/p/chromium/issues/detail?id=1323671&q=pdf%20print%20white%20border&can=2)

I suppose it might be worth checking how other page sizes turn out?

**Comment by G-Ambatte** [comment]
Correct: Windows 10 x64, on Chrome(s) 110.0.5481.178, 113.0.5634.1, and 113.0.5635.0

**Comment by G-Ambatte** [comment]
The only work around that I have found for this is to not use the Chrome Print process, instead using the `Print using System Dialog` option. Experimentally, this did not leave a visible line around the edge.

**Comment by jonathanelscpt** [comment]
also not woking in Ubuntu with Chrome. `Print using System Dialog` doeesn't help.  This is a really basic issue - please provide a fix.

**Comment by Gazook89** [comment]
>  This is a really basic issue

As noted in the initial report, this is a Chrome/Chromium issue.  And it doesn't look like it a priority for the Chromium team.  You can open those links above and star the issues, but that is about as much as can be done.

**Comment by jonathanelscpt** [comment]
But it's not working in Firefox either.

**Comment by Gazook89** [comment]
> But it's not working in Firefox either.

Ah, that is news to me. Unfortunately I can’t get to computer for a few hours to check it out. 

To be sure, the issue you have is a single pixel width line at the bottom of your page?  If it’s any thicker (like an inch), it’s likely the brew is set as a US Letter page size (the default) but the page size when sending to print (to pdf or paper) is set to A4 which is typically the default page size outside of the US. Just mentioning this possibility as that is a quick fix. Otherwise I’ll get online later and check out Firefox.

**Comment by jonathanelscpt** [comment]
I'm on Ubuntu, so may be an edge case.  Chromium and Firefox no good.

I'll retest and update you as well with more specifics.

**Comment by jonathanelscpt** [comment]
Firefox is doing the same

Ubuntu 22.04
firefox 115.0.2 (64-bit


Regards
*Jonathan Els*

**Comment by G-Ambatte** [comment]
A Discord user has reported that setting a `-1px` margin may fix this issue.

`margin:-0px -1px -1px -0px; /*seems for some reason there is a 1px line showing up on print, hard to spot */`

I haven't yet had time to investigate this personally, but I'm recording it here so that it doesn't get lost in the churn.

**Comment by G-Ambatte** [comment]
I believe that this was due to flipping the page using `transform: scaleX(-1)`, but I believe that this has been removed in another PR.

**Comment by 5e-Cleric** [comment]
Can we confirm this issue still exists?

**Comment by roman-dashevich** [comment]
Just exported the PDF on a Mac — same issue. Tried printing from Chrome using the system print dialog, but no luck.
